homeassistant/components/landisgyr_heat_meter/const.py

```diff
@@ -0,0 +1,192 @@
+    ),
+    SensorEntityDescription(
+        key="return_temperature_max_c",
+        name="Return temperature max",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        icon="mdi:thermometer",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="return_temperature_max_previous_year_c",
+        name="Return temperature max previous year",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        icon="mdi:thermometer",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="flow_temperature_max_c",
+        name="Flow temperature max",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        icon="mdi:thermometer",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="flow_temperature_max_previous_year_c",
+        name="Flow temperature max previous year",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        icon="mdi:thermometer",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="operating_hours",
+        name="Operating hours",
+        icon="mdi:clock-outline",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="flow_hours",
+        name="Flow hours",
+        icon="mdi:clock-outline",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="fault_hours",
+        name="Fault hours",
+        icon="mdi:clock-outline",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="fault_hours_previous_year",
+        name="Fault hours previous year",
+        icon="mdi:clock-outline",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="yearly_set_day",
+        name="Yearly set day",
+        icon="mdi:clock-outline",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="monthly_set_day",
+        name="Monthly set day",
+        icon="mdi:clock-outline",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="meter_date_time",
+        name="Meter date time",
+        icon="mdi:clock-outline",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="measuring_range_m3ph",
+        name="Measuring range",
+        native_unit_of_measurement="m3ph",
+        icon="mdi:water-outline",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="settings_and_firmware",
```

Comment:
What kind of value does this return?

Reply:
Its a string that represents some settings and the firmware number. This is added here for diagnostic purpose. Perhaps I could parse it down further in the future, that would certainly be more informative. I'll have to dig into what the number in the string mean exactly though.

Reply:
If it's only for diagnostics, I suggest we remove the sensor and implement a diagnostics platform instead.

Example:
https://github.com/home-assistant/core/blob/dev/homeassistant/components/lametric/diagnostics.py